webapp/security/views.py

```diff
@@ -358,168 +362,223 @@ def cve(cve_id):
 # ===
 
 
-@authorization_required
-def create_cve(cve_id):
-    """
-    Receives a POST request from load_cve.py
-    Parses the object and bulk inserts with add_all()
-    @params json: the body of the request
-    """
+def prepare_cves_to_create(cve_data):
+
+    cve_object = []
+
+    for data in cve_data:
+        packages = []
+        references = []
+        bugs = []
+
+        # Check if there are packages before mapping
+        if data.get("packages"):
+            for package_data in data["packages"]:
+                releases = []
+                for release_data in package_data["releases"]:
+
+                    # Old ignored release
+                    if release_data["name"] == "trusty":
+                        continue
+
+                    # Old ignored release
+                    if release_data["name"] == "precise":
+                        continue
+
+                    # Not showing disco in main table
+                    if release_data["name"] == "disco":
+                        continue
 
-    data = flask.request.json
-    packages = []
-    references = []
-    bugs = []
-
-    # Check if there are packages before mapping
-    if data.get("packages"):
-        for package_data in data["packages"]:
-            releases = []
-            for rel in package_data["releases"]:
-                releases.append(
-                    PackageReleaseStatus(
-                        name=rel["name"],
-                        status=rel["status"],
-                        status_description=rel["status_description"],
+                    # Change and match our releases table
+                    if release_data["name"] == "trusty/esm":
+                        release_data["name"] = "trusty"
+
```

Comment:
We may need to revisit this, once we get from sec team how they are going to update the releases, because we don't always know if ESM has reached end of life or not

Reply:
I don't quite understand this point. We *do* know whether a release is in the ESM phase or not, right? Because we're asking them to submit expiry dates for ESM expiry for releases, right?

It doesn't seem to me that our API should accept "trusty/esm" as a release name, as it's meaningless. "trusty" is the name, "esm" is extra data they use in their system which is irrelevant. We either know "trusty" is in ESM or we don't, regardless of what they submit here.

I think the removing of "esm" should be something done their side, so all they submit is "trusty".

Reply:
No we don't know. Joe sent me this page which indicates some releases are still ESM for certain clients https://wiki.canonical.com/UbuntuEngineering/Security/ESM

Reply:
> Well, we don't know yet. Joe sent me this page which indicates some releases are still ESM for certain clients https://wiki.canonical.com/UbuntuEngineering/Security/ESM, despite the fact they have ended ESM support for quite some time in the standard 5 year period. So these are really the fixes added, so I had to update the releases table. Therefore, we might need a releases API if they are willing to update the ESM dates?

@carkod did Joe tell you that packages in CVEs for expired releases are explicitly considered supported, in the sense of publicly communicating this on the CVE page, while those contracts exists? Yes we have special commercial contracts with some people which means we actually do some support beyond what we publicly promise, but this doesn't seem like it should effect our public data should it?

If we do consider these special contracts in our public information, maybe it could simply be handled by the existing expiry field, in that if we are in practice supporting security for packages in 14.04 until 2023 because of our Ericsson contract, then we could simply set the expiry date as 2023. Or if we support explicit packages in 14.04 because Ericsson asked as to, and we want to expose this publicly, then we would need a "under_special_support" field or something in our packages table.

I guess we need to check with Security which one it is.

> About the last paragraph. I hope that this is the final outcome, but for now the only way to fix the cve/index.html main table, is to fix it our side. I discussed this with @jpmartinspt seems like the most straight-forward way to solve this unpredictable releases they have. If we stick to e.g. trusty, this will cause the package updates to be fail to be recored on our db, because they are updating trusty/esm, not trusty anymore.

You wrote the script you're using to submit CVEs to our API right? It looks to me like that's [here](https://github.com/nottrobin/ubuntu-cve-tracker/pull/2/files#diff-9c88d40408d375390e8bbde08e0ef765R73). You should change this to remove the "ESM", and then change this API so it doesn't accept the "ESM". Any special wrangling of their data should happen in that script, not API side. Our API should be a clean representation of our data model.

Reply:
Ok, that makes sense. I'll make the changes